internal/ingress/controller/template/template.go

```diff
@@ -1343,33 +1343,29 @@ func httpsListener(addresses []string, co string, tc config.TemplateConfig) []st
 	return out
 }
 
-func buildOpentracingForLocation(isOTEnabled bool, location *ingress.Location) string {
+func buildOpentracingForLocation(isOTEnabled bool, isOTTrustSet bool, location *ingress.Location) string {
 	isOTEnabledInLoc := location.Opentracing.Enabled
 	isOTSetInLoc := location.Opentracing.Set
 
 	if isOTEnabled {
 		if isOTSetInLoc && !isOTEnabledInLoc {
 			return "opentracing off;"
 		}
-
-		opc := opentracingPropagateContext(location)
-		if opc != "" {
-			opc = fmt.Sprintf("opentracing on;\n%v", opc)
-		}
-
-		return opc
+	} else if !isOTSetInLoc || !isOTEnabledInLoc {
```

Comment:
Same here, I might be missing something but are we double checking things that might be a legacy behavior?

Reply:
Agreed, in this case we should be able to just check `!isOTEnabledInLoc`. If it's set *and* set to false, don't enable. And if it's not set, the default for this bool would be false anyway. Also don't enable.